edge/key.go

```diff
@@ -122,3 +123,71 @@ func encodeKey(edgeKey *edgeKey) string {
 	encodedKey := base64.RawStdEncoding.EncodeToString([]byte(keyInfo))
 	return encodedKey
 }
+
+func RetrieveEdgeKey(edgeKey string, clusterService agent.ClusterService) (string, error) {
+
+	if edgeKey != "" {
+		log.Println("[INFO] [main,edge] [message: Edge key loaded from options]")
+		return edgeKey, nil
+	}
+
+	var keyRetrievalError error
+
+	edgeKey, keyRetrievalError = retrieveEdgeKeyFromFilesystem()
+	if keyRetrievalError != nil {
+		return "", keyRetrievalError
+	}
+
+	if edgeKey == "" && clusterService != nil {
+		edgeKey, keyRetrievalError = retrieveEdgeKeyFromCluster(clusterService)
+		if keyRetrievalError != nil {
+			return "", keyRetrievalError
+		}
+	}
+
+	return edgeKey, nil
+}
+
+func retrieveEdgeKeyFromFilesystem() (string, error) {
+	var edgeKey string
+
+	edgeKeyFilePath := fmt.Sprintf("%s/%s", agent.DataDirectory, agent.EdgeKeyFile)
+
+	keyFileExists, err := filesystem.FileExists(edgeKeyFilePath)
+	if err != nil {
+		return "", err
+	}
+
+	if keyFileExists {
```

Comment:
I'd say invert this `if` check
```suggestion
	if !keyFileExists {
	   return "", nil
	}
	
	// rest of the function with no indentation
```